src/songuploader/utils.py

```diff
@@ -17,25 +17,30 @@
 from uploader.models import Submission
 import os 
 
-from songuploader.settings import MEDIA_ROOT
-from uploader.models import Submission
-
-
 class ConfiguredLoginViewMixin(LoginRequiredMixin):
     login_url = reverse_lazy("login")
     redirect_field_name = "redirect_to"
 
 
 class LoginRequiredTemplateView(ConfiguredLoginViewMixin, TemplateView):
     template_name: str = None
-    
+
 def download_song(submission: Submission):
-    out_path = os.path.join("/tmp", f"{submission.user.username}.mp4")
-    YouTube(url=submission.song_url).streams.filter(only_audio=True).first().download(output_path=os.path.dirname(out_path), filename=os.path.basename(out_path))
-    open_file = open(out_path, 'rb')
+    mp4_out_path = os.path.join("/tmp", f"{submission.user.username}.mp4")
+    mp3_out_path = os.path.join("/tmp", f"{submission.user.username}.mp3")
+    YouTube(url=submission.song_url).streams.filter(only_audio=True).first().download(output_path=os.path.dirname(mp4_out_path), filename=os.path.basename(mp4_out_path))
+    os.system(f"ffmpeg -i {mp4_out_path} -vn {mp3_out_path}")
+    open_file = open(mp3_out_path, 'rb')
     song_file: File = File(open_file)
     submission.song = song_file
     submission.save()
+    os.remove(mp4_out_path)
+    os.remove(mp3_out_path)
+
+def slice_song(submission: Submission):
+    out_path = os.path.join("/tmp", os.path.basename(submission.song.name))
+    os.system(f"cp {submission.song.path} {out_path}")
+    os.system(f"ffmpeg -ss {submission.start_time} -i {out_path} -c copy -y -t 30 {submission.song.path}")
     os.remove(out_path)
 
```

Comment:
Test if submission has to be updated

Reply:
What exactly. The file name and location saty the same so there is no need to update the submission. There can be some django magic behind it which breaks this stuff but it works right now i can download the song. slice it up using the submission and play the sliced version of it

Reply:
Ok, perfect